Main.cs

```diff
@@ -45,18 +45,25 @@ public List<Result> Query(Query query)
                 if (cachedWindows[query.Search].IsWindow)
                 {
                     var window = cachedWindows[query.Search];
+
                     return new List<Result>
                     {
                         new Result
                         {
                             Title = window.Title,
-                            IcoPath = IconPath,
+                            IcoPath = !String.IsNullOrEmpty(window.Process.Image) ? window.Process.Image: IconPath,
                             Score = cachedWindowsScore,
                             SubTitle = $"{wox_plugin_windowwalker_running} : {window.Process.Name}",
                             ContextData = window,
                             Action = c =>
                             {
+                                OpenWindows.Instance.UpdateFlowWindow();
+
                                 window.SwitchToWindow();
+                                if (OpenWindows.Instance.FlowWindow is not null &&

```

Comment:
Why we want to move the window even if we have switched to it?

Reply:
When using multiple virtual desktops, after moving to a windows on another desktop, the Flow Launcher window remains on the previous desktop.

If you were to activate Flow after switching, you would be brought back to the first desktop.

After switching to a window on another desktop, Flow should switch to that desktop as well. Otherwise you'd be bouncing back and forth between desktops.

Reply:
Thanks! That's pretty nice!

Reply:
Do you think it makes sense to register flow as pinned for all desktop?

Reply:
Not sure how this would behave, pinning is a windows feature.

This seems like something the Flow team needs to figure out, I guess

Reply:
It seems like flow works fine with virtual desktop when using hotkey. Then unless further issue, I would stay it like now.